api/alert_profiles.go

```diff
@@ -0,0 +1,154 @@
+//   if err != nil {
+//     return err
+//   }
+//
+//   alertProfile := api.NewAlertProfile(
+//		"Foo",
+//		api.AlertProfileConfig{
+// 		...
+//     },
+//   )
+//
+//   client.V2.AlertProfiles.Create(AlertProfile)
+//
+func NewAlertProfile(id string, extends string, alerts []AlertProfileAlert) AlertProfileConfig {
+	profile := AlertProfileConfig{
+		Guid:    id,
+		Extends: extends,
+		Alerts:  alerts,
+	}
+	return profile
+}
+
+// List returns a list of Alert Profiles
+func (svc *AlertProfilesService) List() (response AlertProfilesResponse, err error) {
+	err = svc.client.RequestDecoder("GET", apiV2AlertProfiles, nil, &response)
+	return
+}
+
+// Create creates a single Alert Profile
+func (svc *AlertProfilesService) Create(profile AlertProfileConfig) (
+	response AlertProfileResponse,
+	err error,
+) {
+	err = svc.client.RequestEncoderDecoder("POST", apiV2AlertProfiles, profile, &response)
+	return
+}
+
+// Delete deletes a Alert Profile that matches the provided guid
+func (svc *AlertProfilesService) Delete(guid string) error {
+	if guid == "" {
+		return errors.New("specify an intgGuid")
+	}
+
+	return svc.client.RequestDecoder(
+		"DELETE",
+		fmt.Sprintf(apiV2AlertProfileFromGUID, guid),
+		nil,
+		nil,
+	)
+}
+
+// Update updates a single Alert Profile of the provided guid.
+func (svc *AlertProfilesService) Update(guid string, data AlertProfileUpdate) (
+	response AlertProfileResponse,
+	err error,
+) {
+	if guid == "" {
+		err = errors.New("specify a Guid")
+		return
+	}
+	apiPath := fmt.Sprintf(apiV2AlertProfileFromGUID, guid)
+	err = svc.client.RequestEncoderDecoder("PATCH", apiPath, data, &response)
+	return
+}
+
+// Get returns a raw response of the Alert Profile with the matching guid.
+func (svc *AlertProfilesService) Get(guid string, response interface{}) error {
+	if guid == "" {
+		return errors.New("specify a Guid")
+	}
+	apiPath := fmt.Sprintf(apiV2AlertProfileFromGUID, guid)
+	return svc.client.RequestDecoder("GET", apiPath, nil, &response)
+}
+
+type AlertProfile struct {
+	Guid            string                        `json:"alertProfileId,omitempty"`
+	Extends         string                        `json:"extends"`
+	Fields          []AlertProfileField           `json:"fields,omitempty"`
+	DescriptionKeys []AlertProfileDescriptionKeys `json:"descriptionKeys,omitempty"`
+	Alerts          []AlertProfileAlert           `json:"alerts"`
+}
+
+type AlertProfileConfig struct {
+	Guid    string              `json:"alertProfileId"`
+	Extends string              `json:"extends"`
+	Alerts  []AlertProfileAlert `json:"alerts"`
+}
+
+type AlertProfileField struct {
+	Name string `json:"name"`
+}
+
+type AlertProfileDescriptionKeys struct {
+	Name string `json:"name"`
+	Spec string `json:"spec"`
+}
+
+type AlertProfileAlert struct {
+	Name        string `json:"name"`
+	EventName   string `json:"eventName"`
+	Description string `json:"description"`
+	Subject     string `json:"subject"`
+}
+
+type AlertProfileUpdate struct {
```

Comment:
I can't believe that we can't update the Alert Profile itself, this is only updating the alert templates 😕

Reply:
Yes, the api only allows updating the alert templates
https://your-account.lacework.net/api/v2/docs#tag/AlertProfiles/paths/~1api~1v2~1AlertProfiles~1{id}/patch